### ISSUE TITLE
Display issue when deleting assets on IOS app

### ISSUE THREAD

**Issue by junglebells** [immich-app/immich#12361]
### The bug

I am trying to delete a few pictures on immich IOS app. I am also allowing immich to delete it from the photo library. 

The first few deletions go well. After each delete the next picture shows up. However, after 4-8 deletes, the missing picture icon shows up instead of the next picture (see screenshot). There is no way to get rid of it unless you exit the app and try again. 

Can be consistently reproduced. 
![IMG_9900](https://github.com/user-attachments/assets/2664ecf2-af68-4fc1-b03f-a317e205e8b5)

### The OS that Immich Server is running on

Debian 12 latest version 

### Version of Immich Server

1.113.1

### Version of Immich Mobile App

1.113.1

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Default- nothing different.
```

### Your .env content

```Shell
Default - nothing different.
```

### Reproduction steps

1. Login using IOS app
2. Hit the delete icon on bottom right
3. Allow to delete from library 
4. Repeat a few times in succession 
5. After about 4 - 8 deletes, missing picture icon shows up instead of next picture 

this issue has been out there since a long time. I am getting around to report it just now. Thanks for looking into it. 
...

**Comment by alextran1502** [comment]
Can you help get us the mobile app log?

**Comment by junglebells** [comment]
Yes, how do I do that?

**Comment by alextran1502** [comment]
You can open the profile panel and click on the log button, then click on the share button on the upper top right corner to share the logs

<img width="333" alt="image" src="https://github.com/user-attachments/assets/101468bb-6a90-4c89-82e8-8d88be4d9c49">

**Comment by junglebells** [comment]
[Immich_log_2024-09-05T11:58:39.073973.log](https://github.com/user-attachments/files/16899091/Immich_log_2024-09-05T11.58.39.073973.log)

log file attached

**Comment by junglebells** [comment]
Were you able to locate the error in the log file. It happened again today. Here’s the error log entry. Hope this helps. 
![IMG_9924](https://github.com/user-attachments/assets/32e49a04-3c4f-4830-be57-3a56cda88c8a)

**Comment by alextran1502** [comment]
I think the issue is the app is trying to read the information from the cache when it was removed.

**Comment by junglebells** [comment]
That makes sense, however why would it work successfully for the first 4-5 deletes and then fail? Could the asset data be deleted from the cache when performing a delete operation?

**Comment by alextran1502** [comment]
yeah I think that makes sense, so it seems like there are bugs in this mechanism

**Comment by junglebells** [comment]
Got it, thanks Alex. Hopefully you're able to prioritize this issue. Thanks again for your help.

**Comment by junglebells** [comment]
This is still an issue. @alextran1502 - could you please help prioritize. Thanks

**Comment by jrasm91** [comment]
I could not reproduce this myself in the new timeline, so I believe it is resolved. Can you let me know if you are still running into this issue?
